src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -329,4 +310,36 @@ private Map<String, String> extractFromChatParameters(Map<String, String> parame
         }
         return parameters;
     }
+
+    private String parseOutput(String llmOutput, String indexName) {
+        String ppl;
+        Pattern pattern = Pattern.compile("<ppl>((.|[\\r\\n])+?)</ppl>"); // For ppl like <ppl> source=a \n | fields b </ppl>
```

Comment:
Can the `\\r\\n` will match `\n`? Do we have UT to ensure this?

Reply:
The logic is directly from here: https://github.com/opensearch-project/dashboards-assistant/blob/e6359634ff6f532afa983ae0785bc0cd53aee425/server/olly/chains/ppl_generator.ts#L247

But you are right, I don't have UT to cover this line. Will add one.

Reply:
Already add. It works.